### PR TITLE
feat(account-lib): implement send-many-memo builder

### PR DESCRIPTION
**Pull request by alebusse**
- Add a sendmany builder.
- Split contract builder to reuse part of it specific contract builder
- Add constansts file.

[stlx-9236](https://bitgoinc.atlassian.net/browse/STLX-9236)

### CODE REVIEW

File: modules/account-lib/src/coin/stx/abstractContractBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  protected _contractName: string;
+  protected _functionName: string;
+  protected _postConditionMode?: PostConditionMode;
+  protected _postConditions?: PostCondition[];
+  protected _functionArgs: ClarityValue[];
+  private _anchorMode: number;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+    this._anchorMode = 3;
```
Comment (rliberal-simtlix) [inline modules/account-lib/src/coin/stx/abstractContractBuilder.ts:30]: Why here the value is 3? Can we add a doc specifying this value?

Comment (alebusse) [reply modules/account-lib/src/coin/stx/abstractContractBuilder.ts:30]: maybe i can use this https://github.com/stacksgov/sips/blob/main/sips/sip-005/sip-005-blocks-and-transactions.md#transaction-encoding

File: modules/account-lib/test/unit/coin/stx/transactionBuilder/sendmanyBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import should from 'should';
+// import BigNum from 'bn.js';
```
Comment (alebusse) [inline modules/account-lib/test/unit/coin/stx/transactionBuilder/sendmanyBuilder.ts:2]:
```suggestion

```

File: modules/account-lib/src/coin/stx/sendmanyBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  send({ address, amount, memo }: SendParams): this {
+    if (!address || !isValidAddress(address)) {
+      throw new BuildTransactionError('Invalid or missing address, got: ' + address);
+    }
+    if (!amount || !isValidAmount(amount)) {
+      throw new BuildTransactionError('Invalid or missing amount, got: ' + amount);
+    }
+
```
Comment (rliberal-simtlix) [inline modules/account-lib/src/coin/stx/sendmanyBuilder.ts:94]: nit: delete extra line

**Review by jrafael-simtlix — APPROVED** [review]

**Review by rliberal-simtlix — DISMISSED** [review]

**Review by rliberal-simtlix — DISMISSED** [review]

**Review by jrafael-simtlix — DISMISSED** [review]

**Review by tylerlevine — APPROVED** [review]
minor nit, rest looks good to me. I assume this has already been reviewed with the proper stacks domain knowledge?

File: modules/account-lib/src/coin/stx/sendmanyBuilder.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        tupleCV({
+          to: standardPrincipalCV(recipient.address),
+          ustx: uintCV(recipient.amount),
+          memo: bufferCVFromString(recipient.memo || ''),
+        }),
+      ),
+    ),
+  ];
+
+  private sendParamsToPostcondition(sendParams: SendParams[]): PostCondition[] | undefined {
```
[the remark this answers is not in the document]
Comment (alebusse) [reply modules/account-lib/src/coin/stx/sendmanyBuilder.ts:60]: Good catch!

**Review by jrafael-simtlix — APPROVED** [review]

**Review by rliberal-simtlix — APPROVED** [review]

**Review by tylerlevine — APPROVED** [review]

File: modules/account-lib/src/coin/stx/sendmanyBuilder.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    ),
+  ];
+
+  private sendParamsToPostcondition(sendParams: SendParams[]): PostCondition[] {
+    const sum: BigNum = sendParams.reduce((current, next) => current.add(new BigNum(next.amount)), new BigNum(0));
+    return [
+      makeStandardSTXPostCondition(
+        getSTXAddressFromPubKeys(
+          this._fromPubKeys,
+          this._coinConfig.network.type === 'mainnet' ? AddressVersion.MainnetMultiSig : AddressVersion.MainnetMultiSig,
```
Comment (alebusse) [inline modules/account-lib/src/coin/stx/sendmanyBuilder.ts:66]: Wrong network on the second option, it should be `TestnetMultiSig`

**Pull request merged**